src/util/transaction.rs

```diff
@@ -114,3 +114,237 @@ where
     s.serialize_field("vout", &outpoint.vout)?;
     s.end()
 }
+
+pub(super) mod sigops {
+    use crate::chain::{
+        hashes::hex::FromHex,
+        opcodes::{
+            all::{OP_CHECKMULTISIG, OP_CHECKMULTISIGVERIFY, OP_CHECKSIG, OP_CHECKSIGVERIFY},
+            All,
+        },
+        script::{self, Instruction},
+        Transaction, TxOut, Witness,
+    };
+    use std::collections::HashMap;
+
+    /// Get sigop count for transaction. prevout_map must have all the prevouts.
+    pub fn transaction_sigop_count(
+        tx: &Transaction,
+        prevout_map: &HashMap<u32, &TxOut>,
+    ) -> Result<usize, script::Error> {
+        let input_count = tx.input.len();
+        let mut prevouts = Vec::with_capacity(input_count);
+        for idx in 0..input_count {
+            prevouts.push(
+                *prevout_map
+                    .get(&(idx as u32))
+                    .ok_or(script::Error::EarlyEndOfScript)?,
+            );
+        }
+
+        get_sigop_cost(tx, &prevouts, true, true)
+    }
```

Comment:
In the case of coinbase transactions, we don't have a corresponding prevout for the coinbase input, so this alwasy throws an `EarlyEndOfScript` error.

Reply:
Good catch. Yeah, this is the glue code to patch in our HashMap into the Bitcoin Core code, and it doesn't account for coinbase tx...

Reply:
Fixed.